src/main/java/com/softdevsix/api/exceptions/NotFoundError.java

```diff
@@ -0,0 +1,54 @@
+package com.softdevsix.api.exceptions;
+
+/**
+ * Custom exception to represent a 404 Not Found error.
+ * <p>
+ * This exception is used when the requested resource could not be found on the server.
+ * The exception includes a message and an HTTP status code of 404 (Not Found).
+ * </p>
+ */
+public class NotFoundError implements IApiException {
```

Comment:
Why don't review about RuntimeException?

Reply:
The scope of this subtask was specifically focused on providing an error-handling approach for controllers and services. The NotFoundError and BadRequestError implementations are additional examples included as a starting point.

In the future, the necessary error classes will be implemented for endpoints as required.